packages/core/src/Stores/client-store.js

```diff
@@ -412,8 +412,6 @@ export default class ClientStore extends BaseStore {
             () => {
                 if (!this.is_pre_appstore && window.location.pathname === routes.traders_hub) {
                     window.location.href = routes.root;
-                } else if (this.is_pre_appstore && window.location.pathname === routes.root) {
```

Comment:
why did you remove this? we need to redirect to trader's hub if user's trading_hub flag is enabled and is not synced with the localstorage flag

Reply:
This was the problem for the switch account. If youre in dtrader with pre-appstore turned on and you hard refresh, you will be redirected to traders hub. 

So for the header issue, you wont be seeing `Explore Traders Hub` glitch anymore because i added the `is_logging_in` condition there instead

Reply:
after refreshing the page we are getting get_seeting response so we need to check the new value of the this.account_settings.trading_hub with this.is_pre_apstore value, if they are the same, do not redirect user anywhere. I mean a wrapper above line 413

Reply:
Alright i will see to it 🙏